presto-main/src/main/java/io/prestosql/connector/system/QuerySystemTable.java

```diff
@@ -157,12 +158,12 @@ private static Long toMillis(Duration duration)
         return duration.toMillis();
     }
 
-    // TODO (https://github.com/prestosql/presto/issues/5464) map to timestamp with time zone
-    private static Long toTimeStamp(DateTime dateTime)
+    private static Long toTimestampWithTimeZoneMillis(DateTime dateTime)
     {
         if (dateTime == null) {
             return null;
         }
-        return dateTime.getMillis() * MICROSECONDS_PER_MILLISECOND;
+        // dateTime.getZone() is the server zone, should be of no interest to the user
```

Comment:
We used the server time zone for `JmxRecordSetProvider` since that seems like the right thing to return. If the administrator chooses to run their server in a specific time zone, then that's probably what they want to see when looking at system tables. cc @dain

Reply:
If we had an "instant" type, we would map to that. In the absence of "instant" type, we need to fill some zone, an we usually fill in UTC. eg
https://github.com/prestosql/presto/blob/fbd4ca7f6ca1326a077fd06cd33b687cbfe17cf2/presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java#L617
https://github.com/prestosql/presto/blob/91dbe5149ae593fc8f86d36a304f1f7e8dc9175c/presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraRecordCursor.java#L109

Reply:
Also, the problem with server zone is that it may be a political zone.
Presto engine has no problem representing `(point-in-time, zone)` pairs.
However, due to how they are formatted for the client, client can no longer distinguish values within
summer→winter DST change.

```
presto> SELECT
     ->     increment,
     ->     from_unixtime(base + increment * 3600)
     ->         AT TIME ZONE 'UTC'
     ->         AT TIME ZONE 'America/Los_Angeles'
     -> FROM (VALUES(1572766200)) t(base)
     -> CROSS JOIN UNNEST(sequence(0, 4)) u(increment);
     ->
 increment |                    _col1
-----------+---------------------------------------------
         0 | 2019-11-02 23:30:00.000 America/Los_Angeles
         1 | 2019-11-03 00:30:00.000 America/Los_Angeles
         2 | 2019-11-03 01:30:00.000 America/Los_Angeles
         3 | 2019-11-03 01:30:00.000 America/Los_Angeles
         4 | 2019-11-03 02:30:00.000 America/Los_Angeles
(5 rows)
```

Reply:
https://github.com/prestosql/presto/pull/5488